SonarQube.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -57,25 +63,38 @@ public override bool Execute()
 
             AnalysisConfig config = TaskUtilities.TryGetConfig(this.AnalysisConfigDir, new MSBuildLoggerAdapter(this.Log));
             
-            if (config != null)
+            if (config != null && Language != null)
             {
-                AnalyzerSettings settings = config.AnalyzerSettings;
-                if (settings == null)
+                IList<AnalyzerSettings> analyzers = config.AnalyzersSettings;
+                if (analyzers == null)
                 {
                     this.Log.LogMessage(MessageImportance.Low, Resources.AnalyzerSettings_NotSpecifiedInConfig);
                 }
                 else
                 {
-                    this.RuleSetFilePath = settings.RuleSetFilePath;
-
-                    if (settings.AnalyzerAssemblyPaths != null)
-                    {
-                        this.AnalyzerFilePaths = settings.AnalyzerAssemblyPaths.Where(f => IsAssemblyLibraryFileName(f)).ToArray();
+                    bool found = false;
+                    foreach (AnalyzerSettings settings in analyzers) {
```

Comment:
this `foreach` could be replaced by a LINQ query in the form of: `analyzers.SingleOrDefault(settings => Language == settings.Language)`

Note that `FirstOrDefault()` behaves slightly differently